airbyte-webapp/src/components/base/Switch/Switch.tsx

```diff
@@ -0,0 +1,22 @@
+import React from "react";
+
+import styles from "./Switch.module.scss";
+
+interface SwitchProps extends React.InputHTMLAttributes<HTMLInputElement> {
+  small?: boolean;
+  loading?: boolean;
+}
+
+export const Switch: React.FC<SwitchProps> = ({ loading, small, checked, value, ...props }) => {
+  const isChecked = checked || !!value;
+  const labelStyle = `${styles.switch} ${small ? styles.small : ""} ${loading ? styles.loading : ""}`;
+  const spanStyle = `${styles.slider} ${small ? styles.small : ""} ${loading ? styles.loading : ""} ${
+    isChecked ? styles.checked : styles.unchecked
+  }`;
+  return (
+    <label onClick={(event: React.SyntheticEvent) => event.stopPropagation()} className={labelStyle}>
+      <input className={styles.switchInput} type="checkbox" {...props} value={value} checked={isChecked} />
```

Comment:
the props spread should probably go first so that it doesn't override the input type nor the className